ddspipe_core/src/cpp/types/dynamic_types/schema_idl.cpp

```diff
@@ -149,84 +156,89 @@ std::string sequence_kind_to_str(
 }
 
 std::string map_kind_to_str(
-        const fastrtps::types::DynamicType_ptr& dyn_type)
+        const fastdds::dds::DynamicType::_ref_type& dyn_type)
 {
     std::stringstream ss;
-
-    auto key_type = dyn_type->get_descriptor()->get_key_element_type();
-    auto value_type = dyn_type->get_descriptor()->get_element_type();
+    fastdds::dds::TypeDescriptor::_ref_type type_descriptor {fastdds::dds::traits<fastdds::dds::TypeDescriptor>::make_shared()};
+    const auto ret = dyn_type->get_descriptor(type_descriptor);
+    if (ret != fastdds::dds::RETCODE_OK)
+    {
+        throw utils::InconsistencyException("No Type Descriptor");
+    }
+    auto key_type = type_descriptor->key_element_type();
+    auto value_type = type_descriptor->element_type();
     ss << "map<" << type_kind_to_str(key_type) << ", " << type_kind_to_str(value_type) << ">";
 
     return ss.str();
 }
 
 std::string type_kind_to_str(
-        const fastrtps::types::DynamicType_ptr& dyn_type)
+        const fastdds::dds::DynamicType::_ref_type& dyn_type)
 {
     switch (dyn_type->get_kind())
     {
-        case fastrtps::types::TK_BOOLEAN:
+        case fastdds::dds::xtypes::TK_BOOLEAN:
             return "boolean";
 
-        case fastrtps::types::TK_BYTE:
+        case fastdds::dds::xtypes::TK_BYTE:
             return "octet";
 
-        case fastrtps::types::TK_INT16:
+        case fastdds::dds::xtypes::TK_INT16:
```

Comment:
Also TODO: add support for alias, bitset and bitmask types.

Reply:
I also believe it is the right time to move this method to Fast-DDS repo, although it might be messy while we need to support both 3.0 and previous versions.

Reply:
Done in https://github.com/eProsima/Fast-DDS/pull/4787